src/tests/validity/array_get_range_local.ab

```diff
@@ -0,0 +1,129 @@
+// Output
+// Values at -1..0: (0) []
+// Values at -1..1: (1) [zero]
+// Values at -1..2: (2) [zero one]
+// Values at -1..3: (3) [zero one two]
+// Values at -1..4: (3) [zero one two]
+// Values at 0..0: (0) []
+// Values at 0..1: (1) [zero]
+// Values at 0..2: (2) [zero one]
+// Values at 0..3: (3) [zero one two]
+// Values at 0..4: (3) [zero one two]
+// Values at 1..0: (0) []
+// Values at 1..1: (0) []
+// Values at 1..2: (1) [one]
+// Values at 1..3: (2) [one two]
+// Values at 1..4: (2) [one two]
+// Values at 2..0: (0) []
+// Values at 2..1: (0) []
+// Values at 2..2: (0) []
+// Values at 2..3: (1) [two]
+// Values at 2..4: (1) [two]
+// Values at 3..0: (0) []
+// Values at 3..1: (0) []
+// Values at 3..2: (0) []
+// Values at 3..3: (0) []
+// Values at 3..4: (0) []
+// Values at 4..0: (0) []
+// Values at 4..1: (0) []
+// Values at 4..2: (0) []
+// Values at 4..3: (0) []
+// Values at 4..4: (0) []
+// Values at -1..=-1: (0) []
+// Values at -1..=0: (1) [zero]
+// Values at -1..=1: (2) [zero one]
+// Values at -1..=2: (3) [zero one two]
+// Values at -1..=3: (3) [zero one two]
+// Values at 0..=-1: (0) []
+// Values at 0..=0: (1) [zero]
+// Values at 0..=1: (2) [zero one]
+// Values at 0..=2: (3) [zero one two]
+// Values at 0..=3: (3) [zero one two]
+// Values at 1..=-1: (0) []
+// Values at 1..=0: (0) []
+// Values at 1..=1: (1) [one]
+// Values at 1..=2: (2) [one two]
+// Values at 1..=3: (2) [one two]
+// Values at 2..=-1: (0) []
+// Values at 2..=0: (0) []
+// Values at 2..=1: (0) []
+// Values at 2..=2: (1) [two]
+// Values at 2..=3: (1) [two]
+// Values at 3..=-1: (0) []
+// Values at 3..=0: (0) []
+// Values at 3..=1: (0) []
+// Values at 3..=2: (0) []
+// Values at 3..=3: (0) []
+// Values at 4..=-1: (0) []
+// Values at 4..=0: (0) []
+// Values at 4..=1: (0) []
+// Values at 4..=2: (0) []
+// Values at 4..=3: (0) []
+
+fun show_range(label, array) {
+    echo "Values at {label}: ({len(array)}) [{array}]"
+}
+
+main {
+    let array = ["zero", "one", "two"]
+    let slice = array[-1..0]; show_range("-1..0", slice)
```

Comment:
Lets use the nested for loops from "array_get_range_by_ref.ab" to shorten the test.

Reply:
I wrote these out explicitly, to test the compile time arithmetic code. Commented to this effect.